connector_magento_configurable/models/product_attribute_line/importer.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo.addons.component.core import Component
+from odoo.addons.connector.components.mapper import mapping
+from odoo.addons.connector.exception import MappingError
+
+
+class ProductAttributeLineBatchImporter(Component):
+    """ Import the Magento Product Attribute Lines.
+    """
+    _name = 'magento.product.attribute.line.batch.importer'
+    _inherit = 'magento.direct.batch.importer'
+    _apply_on = ['magento.product.attribute.line']
+
+    def _write_product(self, magento_product, tmpl_id, value_ids):
+        magento_product.write(
+            {'product_tmpl_id': tmpl_id,
+             'attribute_value_ids': value_ids})
+        magento_product.odoo_id.write(
+            {'product_tmpl_id': tmpl_id,
+             'attribute_value_ids': value_ids})
+
+    def _get_magento_product_attribute_line(self,
+                                            attribute, value, magento_product):
+        line = {}
+        line['attribute_id'] = attribute['odoo_id'][0]
+        line['value_ids'] = [(4, value.odoo_id.id)]
+        line['template_id'] = magento_product.odoo_id.product_tmpl_id.id
+        line['attribute_name'] = attribute['name']
+        line['external_id'] = str(line['template_id'])
+        line['external_id'] += '_'
+        line['external_id'] += line['attribute_name']
+        return line
+
+    def _import_magento_product_attribute_line(self,
+                                               record, variant,
+                                               attribute, value):
+        line = self._get_magento_product_attribute_line(
+            attribute,
+            value,
+            record
+        )
+        self._import_record(line)
+        self.env['product.template'].search([
+            ('product_variant_ids', '=', False)
+        ]).unlink()
```

Comment:
It seems a bit risky to do that. How do you know you are deleting only what you should? There is for instances addons which changes the default behavior, variants are not created automatically and you have to create them with a button. Would be catastrophic here.